src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -194,6 +236,59 @@ export class ZHANetworkVisualizationPage extends LitElement {
     return label;
   }
 
+  private _handleSearchChange(ev: CustomEvent) {
+    this._filter = ev.detail.value;
+    if (!this._network) {
+      return;
+    }
+    if (this._filter && this._filter !== "") {
+      const filteredNodeIds: (string | number)[] = this._nodes
+        .filter(
+          (node: Node) =>
+            node.label &&
+            node.label.toLowerCase().includes(this._filter!.toLowerCase())
+        )
+        .map((node: Node) => node.id!);
+      this._zoomedDeviceId = "";
+      this._zoomOut();
+      this._network.selectNodes(filteredNodeIds, true);
+    } else {
+      this._network.selectNodes([], false);
+    }
+  }
+
+  private _zoomToDevice(event: PolymerChangedEvent<string>) {
+    event.stopPropagation();
+    this._zoomedDeviceId = event.detail.value;
+    if (!this._network) {
+      return;
+    }
+    this._filter = "";
+    if (!this._zoomedDeviceId || this._zoomedDeviceId === "") {
+      this._zoomOut();
+    } else {
+      const device: ZHADevice | undefined = this._devicesByDeviceId.get(
+        this._zoomedDeviceId
+      );
+      if (device) {
+        this._network.fit({
+          nodes: [device.ieee],
+          animation: { duration: 500, easingFunction: "easeInQuad" },
+        });
+      }
+    }
+  }
+
+  private _zoomOut() {
+    const allNodeIds: (string | number)[] = this._nodes.map(
+      (node: Node) => node.id!
+    );
+    this._network!.fit({
+      nodes: allNodeIds,
```

Comment:
I think you can use fit without `nodes`, will it not zoom out then?

Reply:
looks like it